src/custom/state/lists/reducer/reducerMod.ts

```diff
@@ -0,0 +1,236 @@
+import {
+  // DEFAULT_ACTIVE_LIST_URLS,
+  DEFAULT_LIST_OF_LISTS_BY_NETWORK,
+  DEFAULT_ACTIVE_LIST_URLS_BY_NETWORK,
+  DEFAULT_NETWORK_FOR_LISTS
+} from 'constants/lists'
+import { createReducer } from '@reduxjs/toolkit'
+import { getVersionUpgrade, VersionUpgrade } from '@uniswap/token-lists'
+import { TokenList } from '@uniswap/token-lists/dist/types'
+// import { DEFAULT_LIST_OF_LISTS } from '@src/constants/lists'
+import { updateVersion } from 'state/global/actions'
+import { acceptListUpdate, addList, fetchTokenList, removeList, enableList, disableList } from 'state/lists/actions'
+import { ChainId } from '@uniswap/sdk'
+
+// Mod: change state shape - adds network map
+export type ListsStateByNetwork = {
+  [chain in ChainId]: ListsState
+}
+
+export interface ListsState {
+  readonly byUrl: {
+    readonly [url: string]: {
+      readonly current: TokenList | null
+      readonly pendingUpdate: TokenList | null
+      readonly loadingRequestId: string | null
+      readonly error: string | null
+    }
+  }
+  // this contains the default list of lists from the last time the updateVersion was called, i.e. the app was reloaded
+  readonly lastInitializedDefaultListOfLists?: string[]
+
+  // currently active lists
+  readonly activeListUrls: string[] | undefined
+}
+
+export type ListState = ListsState['byUrl'][string]
+
+export const NEW_LIST_STATE: ListState = {
+  error: null,
+  current: null,
+  loadingRequestId: null,
+  pendingUpdate: null
+}
+
+export type Mutable<T> = { -readonly [P in keyof T]: T[P] extends ReadonlyArray<infer U> ? U[] : T[P] }
+
+// MOD: setInitialState sets each chainId state individually
+const setInitialListState = (chainId: ChainId): ListsState => ({
+  lastInitializedDefaultListOfLists: DEFAULT_LIST_OF_LISTS_BY_NETWORK[chainId],
+  byUrl: {
+    ...DEFAULT_LIST_OF_LISTS_BY_NETWORK[chainId].reduce<Mutable<ListsState['byUrl']>>((memo, listUrl) => {
+      memo[listUrl] = NEW_LIST_STATE
+      return memo
+    }, {})
+  },
+  activeListUrls: DEFAULT_ACTIVE_LIST_URLS_BY_NETWORK[chainId]
+})
+
+const ChainIdList = Object.values(ChainId)
+// MOD: change the intiialState shape
+// we make an object with each chainId pulled from ChainId enum
+// into a list and reduced into a map
+const initialState: ListsStateByNetwork = {
+  // cut in half as enums are always represented as key/value and then inverted
+  // https://stackoverflow.com/a/51536142
+  ...(ChainIdList.slice(ChainIdList.length / 2) as ChainId[]).reduce((memo, chainId) => {
+    if (!memo[chainId]) {
+      memo[chainId] = setInitialListState(chainId)
+    }
+    return memo
+  }, {} as ListsStateByNetwork)
+}
+
+export default createReducer(initialState, builder =>
+  builder
+    .addCase(
+      fetchTokenList.pending,
+      (baseState, { payload: { chainId = DEFAULT_NETWORK_FOR_LISTS, requestId, url } }) => {
```

Comment:
defaults to `constant/index` `DEFAULT_NETWORK_FOR_LISTS` which here = `ChainId.MAINNET`